File: src/main/java/io/myosotisdev/utopianism/modules/craft/Workbench.java
package io.myosotisdev.utopianism.modules.craft;

import io.myosotisdev.utopianism.Locales;
import net.minestom.server.coordinate.Pos;
import net.minestom.server.entity.Player;
import net.minestom.server.inventory.Inventory;
import net.minestom.server.inventory.InventoryType;

public class Workbench
        extends CraftingStation
{
    static final String WorkbenchKey = "workbench";

    public Workbench(boolean shared, Pos pos)
    {
        super(WorkbenchKey, shared, pos);
    }

    public Workbench(String key, boolean shared, Pos pos, String recipesKey)
    {
        super(key, shared, pos, recipesKey);
    }

    @Override
    public Worker createWorker(Player player)
    {
        return new Workbench.Worker(this, player);
    }

    public static class Worker
            extends CraftingStation.Worker
    {
        public Worker(CraftingStation station, Player player)
        {
            super(station, new Inventory(InventoryType.WINDOW_3X3, Locales.get(player.getLocale(), WorkbenchKey + ".title")));
        }
    }
}

File: src/main/java/io/myosotisdev/utopianism/modules/craft/Workbench.kt
package io.myosotisdev.utopianism.modules.craft

import io.myosotisdev.utopianism.Locales
import net.minestom.server.coordinate.Pos
import net.minestom.server.entity.Player
import net.minestom.server.inventory.Inventory
import net.minestom.server.inventory.InventoryType

class Workbench : CraftingStation
{
    constructor(shared: Boolean, pos: Pos?) : super("workbench", shared, pos!!)
    {
    }

    constructor(shared: Boolean, pos: Pos?, recipesKey: String?) : super("workbench", shared, pos!!, recipesKey)
    {
    }

    override fun createWorker(player: Player?): Worker
    {
        return Worker(this, Inventory(InventoryType.WINDOW_3X3, Locales.translate(player!!.locale, "utopianism:craft.workbench.title")))
    }
}